scripts/qa/section-order.php

```diff
@@ -177,8 +177,8 @@ function checkSectionErrors(string $path): array
 
     /* Constructors are special */
     if (str_ends_with($path, 'construct.xml')) {
-        if (!str_contains($content, '<constructorsynopsis>')
-            && !str_contains($content, '<constructorsynopsis role="oop">')
+        if (!str_contains($content, '<constructorsynopsis>') &&
+            !preg_match('/<constructorsynopsis role=".*">/', $content)
```

Comment:
Oh, wait, we shouldn't do greedy matching:
```suggestion
            !preg_match('/<constructorsynopsis role="[^"]*">/', $content)
```
Same below.